### PR TITLE
[MZvault conversion] updates

### PR DESCRIPTION
**Pull request by robinschmid**
Done and worked on the test file.

### Ideas
Currently, I only copied logging_utils.py - its a duplicate from the FBMN workflow.
It would b nice if we could just move it to shared_code and then define this folder as a source folder to all the other projects / files. No idea how you do this on this scale though.

### Change
library_conversion.py takes more parameters now:
- PI, data collector and
- library format (mzvault, nist, or this bmdms what ever this is)

Plus I added error handling to the requests:
- SMILES -> InChI
- SMILES -> exact mass

the output is something like this - if the input SMILES is "3" (invalid):
```log
2021-05-16 15:39:48,798 - __main__ - WARNING - Failed to get InChI from SMILES at: https://gnps-structure.ucsd.edu/inchi?smiles=3
2021-05-16 15:39:48,799 - __main__ - ERROR - 400 Client Error: BAD REQUEST for url: https://gnps-structure.ucsd.edu/inchi?smiles=3
Traceback (most recent call last):
  File "D:\git\GNPS_Workflows\library_conversion\tools\library_conversion\MyTest.py", line 21, in <module>
    inchi_convert_results.raise_for_status()
  File "C:\Program Files\Python39\lib\site-packages\requests\models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: BAD REQUEST for url: https://gnps-structure.ucsd.edu/inchi?smiles=3
```

### CODE REVIEW

File: library_conversion/tools/library_conversion/nist_msp_conversion.py [hunk 1]
```diff
@@ -1,26 +1,24 @@
 #!/usr/bin/python
```
Comment (mwang87) [inline library_conversion/tools/library_conversion/nist_msp_conversion.py:1]: This file should not be included

File: search_single_spectrum/test/test_foodomics.py [hunk 2]
```diff
@@ -9,9 +9,9 @@ def test():
     input_filename = "data/foodomics/all_dataset_matchs.tsv"
```
Comment (mwang87) [inline search_single_spectrum/test/test_foodomics.py:9]: why is this change in this PR?

File: library_conversion/tools/library_conversion/library_conversion.py [hunk 3]
```diff
@@ -3,15 +3,41 @@
...
-import mzvault_conversion as mzvault_convert
+from enum import Enum
+
+
+# Specifies the input formats
+class InputFormat(Enum):
+    mzvault, nist_msp, bmdms = range(3)
+
+# handle formats and convert
+def convert(libformat, input_filename, mgf_filename, batch_filename, pi_name, collector_name):
```
Comment (mwang87) [inline library_conversion/tools/library_conversion/library_conversion.py:14]: We do not need these other formats. This is only meant for mzvault.

File: search_single_spectrum/test/test_foodomics.py (hunk 2 above)
Comment (robinschmid) [reply search_single_spectrum/test/test_foodomics.py:9]: I updated the branch to the latest master to ensure compatibility. This PR merges into the other library conversion workflow branch that is currently not up to date with the master.

File: library_conversion/tools/library_conversion/library_conversion.py (hunk 3 above)
Comment (robinschmid) [reply library_conversion/tools/library_conversion/library_conversion.py:14]: I added a test file for other msp formats that are compatible with the NIST specs.
This format is also used by MZmine and other tools to create libraries.

File: library_conversion/tools/library_conversion/nist_msp_conversion.py (hunk 1 above)
Comment (robinschmid) [reply library_conversion/tools/library_conversion/nist_msp_conversion.py:1]: I added a test file for other msp formats that are compatible with the NIST specs.
This format is also used by MZmine and other tools to create libraries.

**Comment by mwang87** [comment]
Can you link an example test in Proteomics2 with an MSP file for the NIST format parser

**Comment by robinschmid** [comment]
I now also added the MZmine json format that we use for the direct submission of library entries from MZmine to GNPS,
So far I have created unit tests in the workflow and added test data for each file format.

**Comment by robinschmid** [comment]
### Fixes
- simple sanity checks before adding entries (compound name not empty)
- exception on empty output 
- exception on wrong input

### Input
- added combo box for input format
- text fields for PI name and data collector name (optional)

### Tests
- test different file formats
- test wrong input file for format selection (e.g., MZmine json in MZvault msp) -> expected exception
- test empty input file -> expected exception

**Pull request merged**